### PR TITLE
geolocation: use maxmind

### LINKED ISSUE #140: location: support ip6
Consider switching to Maxmind

### PR DESCRIPTION
**Pull request by jsign**
This PR:
- Add support for IP6 resolution
- Switch to using a new offline geolocation free database (MaxMind) which seems more reliable and doesn't have a deprecated client.
- Switched to ISO Codes for countries, sound better to use a standard.
- Moved `Dockerfile` to the root of the repository.

Bottom line: in the current interop, the previous geolocation db resolved 2 of 13 miners' location. The current PR resolved 13 of 13.

<details><summary>Stripped current index:</summary>
<p>

```
"Meta": {
    "Online": 13,
    "Offline": 74,
    "Info": {
      "t01812": {
        "LastUpdated": "2020-06-15T15:17:21.202388799Z",
        "UserAgent": "lotus-0.4.0+git.96a5e579.dirty",
        "Location": {
          "Country": "CN",
          "Longitude": 113.7266,
          "Latitude": 34.7725
        },
        "Online": true
      },
      "t01841": {
        "LastUpdated": "2020-06-15T15:17:21.20237101Z",
        "UserAgent": "lotus-0.4.0+git.e473f629",
        "Location": {
          "Country": "CN",
          "Longitude": 113.7266,
          "Latitude": 34.7725
        },
        "Online": true
      },
      "t01843": {
        "LastUpdated": "2020-06-15T15:17:21.202094157Z",
        "UserAgent": "lotus-0.4.0+git.90665889",
        "Location": {
          "Country": "CN",
          "Longitude": 112.3554,
          "Latitude": 25.4015
        },
        "Online": true
      },
      "t01845": {
        "LastUpdated": "2020-06-15T15:17:21.201833097Z",
        "UserAgent": "lotus-0.4.0+git.e473f629",
        "Location": {
          "Country": "CN",
          "Longitude": 113.7266,
          "Latitude": 34.7725
        },
        "Online": true
      },
      "t01908": {
        "LastUpdated": "2020-06-15T15:17:21.201708353Z",
        "UserAgent": "lotus-0.4.0.5+git.0418441f.dirty",
        "Location": {
          "Country": "HK",
          "Longitude": 114.15,
          "Latitude": 22.2909
        },
        "Online": true
      },
      "t02094": {
        "LastUpdated": "2020-06-15T15:17:21.200960705Z",
        "UserAgent": "lotus-0.4.0+git.e473f629",
        "Location": {
          "Country": "CN",
          "Longitude": 113.7266,
          "Latitude": 34.7725
        },
        "Online": true
      },
      "t02285": {
        "LastUpdated": "2020-06-15T15:17:21.202223678Z",
        "UserAgent": "lotus-0.4.0+git.e473f629",
        "Location": {
          "Country": "CN",
          "Longitude": 113.7266,
          "Latitude": 34.7725
        },
        "Online": true
      },
      "t02293": {
        "LastUpdated": "2020-06-15T15:17:21.201642528Z",
        "UserAgent": "lotus-0.4.0+git.e473f629",
        "Location": {
          "Country": "CN",
          "Longitude": 113.7266,
          "Latitude": 34.7725
        },
        "Online": true
      },
      "t02407": {
        "LastUpdated": "2020-06-15T15:17:21.202265554Z",
        "UserAgent": "lotus-0.4.0+git.6b4208f.dirty",
        "Location": {
          "Country": "CN",
          "Longitude": 113.7266,
          "Latitude": 34.7725
        },
        "Online": true
      },
      "t02418": {
        "LastUpdated": "2020-06-15T15:17:21.201255646Z",
        "UserAgent": "lotus-0.4.0+git.90665889",
        "Location": {
          "Country": "CN",
          "Longitude": 108.9294,
          "Latitude": 34.2598
        },
        "Online": true
      },
      "t02430": {
        "LastUpdated": "2020-06-15T15:17:21.202304311Z",
        "UserAgent": "lotus-0.4.0+git.e473f629",
        "Location": {
          "Country": "CN",
          "Longitude": 113.7266,
          "Latitude": 34.7725
        },
        "Online": true
      },
      "t02541": {
        "LastUpdated": "2020-06-15T15:17:21.201116928Z",
        "UserAgent": "lotus-0.4.0+git.90665889",
        "Location": {
          "Country": "CN",
          "Longitude": 113.7266,
          "Latitude": 34.7725
        },
        "Online": true
      },
      "t02544": {
        "LastUpdated": "2020-06-15T15:17:21.200846335Z",
        "UserAgent": "lotus-0.4.0+git.96a5e579.dirty",
        "Location": {
          "Country": "CN",
          "Longitude": 108.3146,
          "Latitude": 22.8137
        },
        "Online": true
      }
    }
  },
```
</p>
</details>

Closes #140

### CODE REVIEW

File: api/client/utils_test.go [hunk 1]
```diff
@@ -50,6 +50,7 @@ func setupServer(t *testing.T) func() {
 		GrpcWebProxyAddress: grpcWebProxyAddress,
 		RepoPath:            repoPath,
 		GatewayHostAddr:     gatewayHostAddr,
+		MaxMindDBFolder:     "../../iplocation/maxmind",
```
Comment (jsign) [inline api/client/utils_test.go:53]: This new `MaxMindDBFolder` is mostly for integration testing purposes. The default value (empty string) is the correct one for the baked DockerHub image, so no need to tune anything.

File: docker/docker-compose.yaml [hunk 2]
```diff
@@ -10,7 +10,6 @@ services:
   powergate:
     build:
       context: ../
-      dockerfile: cmd/powd/Dockerfile
```
Comment (jsign) [inline docker/docker-compose.yaml:13]: Since now `Dockerfile` lives on the root, no need for this.

File: index/miner/types.go [hunk 3]
```diff
@@ -42,6 +42,6 @@ type Meta struct {
 // Location contains geeoinformation.
 type Location struct {
 	Country   string
-	Longitude float32
-	Latitude  float32
+	Longitude float64
```
Comment (jsign) [inline index/miner/types.go:45]: MaxMind lat and long have `float64` precision.

File: iplocation/maxmind/maxmind_test.go [hunk 4]
```diff
@@ -0,0 +1,62 @@
+package maxmind
+
+import (
+	"testing"
+
+	"github.com/multiformats/go-multiaddr"
+	"github.com/stretchr/testify/require"
+)
+
+func TestResolve(t *testing.T) {
```
Comment (jsign) [inline iplocation/maxmind/maxmind_test.go:10]: Added a test with different cases just in case.

File: util/util.go [hunk 5]
```diff
@@ -16,35 +16,46 @@ var (
 )
 
 // TCPAddrFromMultiAddr converts a multiaddress to a string representation of a tcp address.
-func TCPAddrFromMultiAddr(maddr ma.Multiaddr) (addr string, err error) {
+func TCPAddrFromMultiAddr(maddr ma.Multiaddr) (string, error) {
```
Comment (jsign) [inline util/util.go:19]: Basically, support IP6 decoding.

File: net/rpc/rpc.go [hunk 6]
```diff
@@ -89,17 +89,22 @@ func (a *RPC) FindPeer(ctx context.Context, req *FindPeerRequest) (*FindPeerResp
 		addrs[i] = addr.String()
 	}
 
+	var l *Location
+	if peerInfo.Location != nil {
```
Comment (asutula) [inline net/rpc/rpc.go:93]: Nice catch

**Pull request merged**